File: src/main/java/com/inventario/casaempeno/model/Cliente.java
package com.inventario.casaempeno.model;


import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

@Entity
@Table(name = "clientes")
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String nombreCompleto;

    @Email
    @NotBlank
    @Column(unique = true)
    private String correo;

    @NotBlank
    private String contrasena;

    private String estadoFinanciero;

    private String telefono;

    // Getters y Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getNombreCompleto() { return nombreCompleto; }
    public void setNombreCompleto(String nombreCompleto) { this.nombreCompleto = nombreCompleto; }

    public String getCorreo() { return correo; }
    public void setCorreo(String correo) { this.correo = correo; }

    public String getContrasena() { return contrasena; }
    public void setContrasena(String contrasena) { this.contrasena = contrasena; }

    public String getEstadoFinanciero() { return estadoFinanciero; }
    public void setEstadoFinanciero(String estadoFinanciero) { this.estadoFinanciero = estadoFinanciero; }

    public String getTelefono() { return telefono; }
    public void setTelefono(String telefono) { this.telefono = telefono; }

}

File: frontend/lib/models/cliente.dart
class Cliente {
  final int? id;
  final String nombreCompleto;
  final String correo;
  final String contrasena;
  final String estadoFinanciero;
  final String telefono;

  Cliente({
    this.id,
    required this.nombreCompleto,
    required this.correo,
    required this.contrasena,
    required this.estadoFinanciero,
    required this.telefono,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) {
    return Cliente(
      id: json['id'],
      nombreCompleto: json['nombreCompleto'] ?? '',
      correo: json['correo'] ?? '',
      contrasena: json['contrasena'] ?? '',
      estadoFinanciero: json['estadoFinanciero'] ?? '',
      telefono: json['telefono'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'nombreCompleto': nombreCompleto,
      'correo': correo,
      'contrasena': contrasena,
      'estadoFinanciero': estadoFinanciero,
      'telefono': telefono,
    };
  }
}
